pkg/action/bootentries.go

```diff
@@ -229,11 +231,28 @@ func bootNameToSystemdConf(name string) (string, error) {
 // and prompts the user to select one
 // then calls the underlying SelectBootEntry function to mange the entry writing and validation
 func listBootEntriesSystemd(cfg *config.Config) error {
+	var err error
+	e := elemental.NewElemental(cfg)
+	cleanup := utils.NewCleanStack()
+	defer func() { err = cleanup.Cleanup(err) }()
 	// Get EFI partition
 	efiPartition, err := partitions.GetEfiPartition()
 	if err != nil {
 		return err
 	}
+	// mount if not mounted
+	if mounted, err := utils.IsMounted(cfg, efiPartition); !mounted && err == nil {
+		if efiPartition.MountPoint == "" {
+			efiPartition.MountPoint = cnst.EfiDir
+		}
+		err = e.MountPartition(efiPartition)
+		if err != nil {
+			cfg.Logger.Errorf("could not mount EFI partition: %s", err)
+			return err
+		}
+		cleanup.Push(func() error { return e.UnmountPartition(efiPartition) })
+	}
```

Comment:
log the errors if any?

Reply:
That gets auto logged by the clean function IIRC?

Reply:
but this is inside the clean stack, and that auto reports errors I think. This is done everywhere and IIRC always works at the end, it will report the errors while executing the clean function  stack